server/workflowarchive/archived_workflow_server.go

```diff
@@ -299,12 +303,28 @@ func (w *archivedWorkflowServer) RetryArchivedWorkflow(ctx context.Context, req
 			}
 		}
 
+		log.WithFields(log.Fields{"Dehydrate workflow uid=": wf.UID}).Info("RetryArchivedWorkflow")
+		err = w.hydrator.Dehydrate(wf)
+		if err != nil {
+			return nil, sutils.ToStatusError(err, codes.Internal)
+		}
+
 		wf.ObjectMeta.ResourceVersion = ""
 		wf.ObjectMeta.UID = ""
 		result, err := wfClient.ArgoprojV1alpha1().Workflows(req.Namespace).Create(ctx, wf, metav1.CreateOptions{})
 		if err != nil {
 			return nil, sutils.ToStatusError(err, codes.Internal)
 		}
+		if !w.hydrator.IsHydrated(wf) {
```

Comment:
I assume this if statement is just in case? there isn't any reason why it shouldn't be dehydrated, right?

Reply:
(nothing wrong with that of course...just trying to make sure I understand)

Reply:
In line 307, we call the func  w.hydrator.Dehydrate(wf) to Dehydrate the wf, but within this function, it will automatically determine whether(according to the size of this wf, https://github.com/argoproj/argo-workflows/blob/main/workflow/hydrator/hydrator.go#L100) dehydration really needs to be performed, so we should to add this if statement . @juliev0

Reply:
got it. Okay, please add some comments around all of this code, because I think it will be hard otherwise for people to understand the purpose.